inc/class-rewrite.php

```diff
@@ -159,4 +171,34 @@ public function category_rewrite_rules() {
 
 		return $category_rewrite;
 	}
+	/**
+	 * Convert encoded URI string to uppercase.
+	 * 
+	 * @return string
+	 */
+	public function encodedToUpper( &$encoded ) {
+
+		if ( strpos( $encoded, '%' ) !== false  ) {
+	
+			$encoded = strtoupper( $encoded );
+	
+		}
+	
+		return $encoded;
+	
+	}
+	/**
+	 * Walk through category nicename and convert encoded parts
+	 * into uppercase using $this->encodedToUpper().
+	 * 
+	 * @return string
+	 */
+	public function filteredNicename( $name ) {
+    
+		$name = explode( '/', $name );
+	
+		array_walk( $name, [ $this, 'encodedToUpper'] );
```

Comment:
This short array notation is only available from php 5.4 and higher. Can you change this  in the ancient way (`array(...)`), because we the minimal php version we support is 5.2.